stdlib/array.gr

```diff
@@ -92,6 +92,24 @@ export let mapi = (fn, array) => {
   })
 }
 
+export let reduce = (fn, initial, array) => {
+  let acc = box(initial);
```

Comment:
Why not `let mut` here?

Reply:
I think I was getting a type bug with the `acc` value being mutable when passed into the function. Maybe I was doing something wrong?

Reply:
I was doing something wrong. Just pushed the `let mut` changes.